### PR TITLE
rouge: create vfat filesystem with proper sector size

### PR DESCRIPTION
**Pull request by sa-kib**
Fix regression from custom sector size feature -- vfat FS has to be created with correct sector size provided, otherwise it can't be mounted on running target system.

### CODE REVIEW

File: moulin/rouge/ext_utils.py [hunk 1]
```diff
@@ -69,9 +69,9 @@ def mkext4fs(file_out: BinaryIO, contents_dir=None):
     _run_cmd(args)
 
 
-def mkvfatfs(file_out: BinaryIO):
-    "Create ext4 fs in given file"
-    args = ["mkfs.vfat", file_out.name]
+def mkvfatfs(file_out: BinaryIO, sector_size: int):
+    "Create FAT fs in given file"
+    args = ["mkfs.vfat", f"-S {sector_size}", file_out.name]
```
Comment (lorc) [inline moulin/rouge/ext_utils.py:74]: This should be the optional parameter. We don't want to force default value there.

File: moulin/rouge/block_entry.py [hunk 2]
```diff
@@ -313,6 +315,10 @@ def write(self, fp, offset):
 class Vfat(FileSystem):
     "Represents vfat fs with list of files"
 
+    def __init__(self, node: YamlValue):
+        super(Vfat, self).__init__(node)
+        self._sector_size: int = 0
```
Comment (lorc) [inline moulin/rouge/block_entry.py:320]: `__init__` should initialize the whole object, along with a sector size.

I am proposing to move `sector_size` parameter into `kwargs`

File: moulin/rouge/block_entry.py [hunk 3]
```diff
@@ -62,6 +62,8 @@ def __init__(self, node: YamlValue):
         for part_id, part in node["partitions"].items():
             label = part_id
             entry_obj, gpt_type, gpt_guid = self._process_entry(part)
+            if isinstance(entry_obj, Vfat):
```
Comment (lorc) [inline moulin/rouge/block_entry.py:65]: Okay, this is not a good idea. It is better to pass sector size as a keyword argument to an entry constructor (see below).  So, each `BlockEntry` implementation could make own judgement if and how to use this parameter

File: moulin/rouge/ext_utils.py (hunk 1 above)
Comment (sa-kib) [reply moulin/rouge/ext_utils.py]: done

File: moulin/rouge/block_entry.py (hunk 2 above)
Comment (sa-kib) [reply moulin/rouge/block_entry.py]: yes, moved

File: moulin/rouge/block_entry.py (hunk 3 above)
Comment (sa-kib) [reply moulin/rouge/block_entry.py]: fixed

**Review by lorc — APPROVED** [review]
Thanks!

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**